src/transformers/training_args.py

```diff
@@ -1748,15 +1748,21 @@ def __post_init__(self):
                     " `--accelerator_config {'dispatch_batches':VALUE} instead",
                     FutureWarning,
                 )
-                self.accelerator_config["dispatch_batches"] = self.dispatch_batches
+                if isinstance(self.accelerator_config, dict):
```

Comment:
I think a better way to handle this would be make sure `self.accelerator_config` is of a single type -- `AcceleratorConfig` -- rather than have this if/else structure. Otherwise, every time we want to handle the config we'll need to do these if/else checks, creating 4 lines of code (and possible bugs) rather than one

Reply:
Was thinking the same, will fix